src/transformStateWithClones.js

```diff
@@ -7,7 +7,28 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let stateClone = { ...state };
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(stateClone, obj.extraData);
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const keys of obj.keysToRemove) {
+        delete stateClone[keys];
+      }
+    }
+
+    if (obj.type === 'clear') {
+      stateClone = {};
+    }
```

Comment:
Use switch statement if you have limited amount of conditions.
switch/case should always have default case for error handling